app/forms/concerns/wizard.rb

```diff
@@ -0,0 +1,61 @@
+module Wizard
+  extend ActiveSupport::Concern
+
+  included do
+    attr_writer :state_store
+    attr_accessor :current_step, :review_mode
+  end
+
+  def initialize(state_store, params = {})
+    @state_store = state_store
+    sanitize_attributes(params)
+
+    super(last_saved_state.deep_merge(params))
+
+    update_review_mode if current_step.eql?('check')
+  end
+
+  def next_step
+    :check
+  end
+
+  def previous_step
+    :new
+  end
+
+  def clear_state!
+    @state_store.delete
+  end
+
+  alias_method :save, :clear_state!
+
+  def save_state!
+    @state_store.write(state)
+  end
+
+  def valid?
```

Comment:
fwiw I prefer `valid_for_current_step?` but happy to hear other opinions!

Reply:
This overrides active record's `valid?` which is what we call to check for the validity of any object. As the form knows what step we are currently are at, I think it makes sense sto be consistent across the app in our validation calls rather than have to call another method that does the same thing in certain places.

Reply:
I accept that argument. I offer a counterargument, which is that the controller has to accommodate the wizard in various ways too - eg implementing the mapping from next_step and previous_step to routes. I think there might be a case that it's a useful sign that we're dealing with a wizard?

Reply:
Maybe enable both then. In two step wizards where you only have one input step and one verification step (e.g. interviews) there is only one step that can be validated, so in that context `valid_for_current_step` can be confusing.

Reply:
My 2p: while just having a `valid?` method is neater, it reads like the entire wizard instance is valid (ie. either for all steps or all completed steps) rather than for a particular step, so I think it introduces ambiguity from the caller's perspective. 
It also seems like we can't ever now ask if a different step to the current one is valid (not sure we have a use case for this though).